src/TADA_ViewATTAINS.R

```diff
@@ -0,0 +1,157 @@
+        urltools::param_set(key = "returnTrueCurves", value = "false") %>%
+        urltools::param_set(key = "returnIdsOnly", value = "false") %>%
+        urltools::param_set(key = "returnCountOnly", value = "false") %>%
+        urltools::param_set(key = "returnZ", value = "false") %>%
+        urltools::param_set(key = "returnM", value = "false") %>%
+        urltools::param_set(key = "returnDistinctValues", value = "false") %>%
+        urltools::  param_set(key = "returnExtentOnly", value = "false") %>%
+        urltools::param_set(key = "featureEncoding", value = "esriDefault")
+      
+      final_sf <- geojsonsf::geojson_sf(query)
+      
+      return(final_sf)
+      
+    }
+    
+    # grab the geospatial data!
+    nearby_catchments <- download_attains(baseurl = "https://gispub.epa.gov/arcgis/rest/services/OW/ATTAINS_Assessment/MapServer/3/query?") %>%
+      # remove unnecessary columns:
+      dplyr::select(-c(OBJECTID, GLOBALID)) %>%
+      .[TADA_DataRetrieval_data,] %>%
+      dplyr::rename_with(~ paste0("ATTAINS.", .), dplyr::everything())
+    
+    # join TADA sf features to the ATTAINS catchment feature(s) they land on:
+    TADA_with_ATTAINS <- TADA_DataRetrieval_data %>%
+      sf::st_join(., nearby_catchments) %>%
+      sf::st_drop_geometry() 
+    
+    # POINT FEATURES
+    try(points <- download_attains(baseurl = "https://gispub.epa.gov/arcgis/rest/services/OW/ATTAINS_Assessment/MapServer/0/query?") %>%
+          .[nearby_catchments,])
+    try(points_mapper <- tibble::tibble(assessmentunitidentifier = unique(lines$assessmentunitidentifier),
+                                        colors = inferno(length(unique(lines$assessmentunitidentifier)))) %>%
+          dplyr::inner_join(points, by = "assessmentunitidentifier") %>%
+          sf::st_as_sf())
+    
+    # LINE FEATURES
+    try(lines <- download_attains(baseurl = "https://gispub.epa.gov/arcgis/rest/services/OW/ATTAINS_Assessment/MapServer/1/query?") %>%
+          .[nearby_catchments,] %>%
+          distinct(assessmentunitidentifier, .keep_all = TRUE))
+    try(lines_mapper <- tibble(assessmentunitidentifier = unique(lines$assessmentunitidentifier),
+                               colors = viridis::inferno(length(unique(lines$assessmentunitidentifier)))) %>%
+          dplyr::inner_join(lines, by = "assessmentunitidentifier") %>%
+          sf::st_as_sf())
+    
+    # AREA FEATURES
+    try(areas <- download_attains(baseurl = "https://gispub.epa.gov/arcgis/rest/services/OW/ATTAINS_Assessment/MapServer/2/query?") %>%
+          .[nearby_catchments,])
+    try(areas_mapper <- tibble::tibble(assessmentunitidentifier = unique(areas$assessmentunitidentifier),
+                                       colors = viridis::inferno(length(unique(areas$assessmentunitidentifier)))) %>%
+          dplyr::inner_join(areas, by = "assessmentunitidentifier") %>%
+          sf::st_as_sf())
+    
+    sumdat <- data  %>% dplyr::group_by(MonitoringLocationIdentifier, 
```

Comment:
For high-level organization, could this (lines 94-123) be moved up before the point/line/area features, and then put the add points/lines/areas (124-150) inside the try() where each item is downloaded (69-92)?

Then add some error handling via tryCatch() instead of try() so there are messages of "No point/area/line associated with these data" or the like instead of messaging these errors:
![Screenshot 2024-02-05 at 10 01 44 AM](https://github.com/rossyndicate/TADA_funx/assets/32140074/0493948c-2333-4271-93de-3d9a2d63b579)

Reply:
I'm deploying @juandlt-csu for better error handling! :)